### ISSUE TITLE
ErrorException: Uninitialized string offset 1

### ISSUE THREAD

**Issue by fuale** [zircote/swagger-php#1016]
at 4.0.1
```
  at vendor/zircote/swagger-php/src/Analysers/TokenScanner.php:128
    124▕                             // no function body
    125▕                             $this->skipTo($tokens, ';');
    126▕                         }
    127▕ 
  ➜ 128▕                         $units[$currentName]['methods'][] = $token[1];
    129▕                     }
    130▕                     break;
    131▕                 case T_VARIABLE:
    132▕                     if (1 == count($stack) && $currentName) {
```

Place where i found bug using debugger

```
$worklogs->issues = array_map(static function ($issue) use ($client) {
    if ($issue->fields->worklog->total > $issue->fields->worklog->maxResults) {
        $issue->fields->worklog = JiraQuery::getWorklogs($client, $issue->key);
    }

    return $issue;
}, $worklogs->issues);

return collect($worklogs->issues);
```
I Do something wrong?

**Comment by fuale** [comment]
I think it's a problem with nested `static function` statement

**Comment by fuale** [comment]
additional debug info
```
$lastToken = {array} [3]
 0 = {int} 321
 1 = "static"
 2 = {int} 64
$namespace = "App\Services\Jira"
$stack = {array} [1]
 0 = "{"
$token = "("
```

**Comment by fuale** [comment]
Yes, I commented out this piece of code, and now it's working

**Comment by DerManoMann** [comment]
Sigh, thanks. Another test case, another release :/

**Comment by DerManoMann** [comment]
Hmm, there must be something else. I've added this methot to one of the test fixtures and that works fine.
```
    public function whatever($client)
    {
        $worklogs = (object)[];
        $worklogs->issues = array_map(static function ($issue) use ($client) {
            if ($issue->fields->worklog->total > $issue->fields->worklog->maxResults) {
                $issue->fields->worklog = 'foo';
            }

            return $issue;
        }, $worklogs->issues);

        return collect($worklogs->issues);
    }

```

@fuale Would you be able to extract a code snippet into a separate file or something that does break the code and share that? There must be some more context that is missing to make it go wrong.

**Comment by DerManoMann** [comment]
You can see the code I used for a fixture here: 

https://github.com/DerManoMann/swagger-php/blob/more-nested-functions/tests/Fixtures/PHP/AnonymousFunctions.php#L56

**Comment by fuale** [comment]
oh, my fault
```
public function whatever($client)
{
    foreach ($prepareParams as $key => $value) {
        $jql = preg_replace("/:${key}/", $value, $jql);
    }

    $worklogs = JiraQuery::runQuery($client, $jql, [], ['worklogs']);

    $worklogs->issues = array_map(static function ($issue) use ($client) {
        if ($issue->fields->worklog->total > $issue->fields->worklog->maxResults) {
            $issue->fields->worklog = JiraQuery::getWorklogs($client, $issue->key);
        }

        return $issue;
    }, $worklogs->issues);

    return collect($worklogs->issues);
}
```

gives me

```
➜ ./bin/phpunit --filter TokenScannerTest
PHPUnit 9.5.10 by Sebastian Bergmann and contributors.

.........E                                                        10 / 10 (100%)

Time: 00:00.008, Memory: 10.00 MB

There was 1 error:

1) OpenApi\Tests\Analysers\TokenScannerTest::testScanFile with data set "AnonymousFunctions" ('PHP/AnonymousFunctions.php', array(array(array('OpenApi\Annotations\Info'), array(), array(), array('index', 'query', 'other', 'shortFn', 'withUse', 'whatever'), array())))
Uninitialized string offset 1

/home/w/Projects/swagger-php/src/Analysers/TokenScanner.php:128
/home/w/Projects/swagger-php/src/Analysers/TokenScanner.php:21
/home/w/Projects/swagger-php/tests/Analysers/TokenScannerTest.php:182

ERRORS!
Tests: 10, Assertions: 19, Errors: 1.
```

**Comment by fuale** [comment]
@DerManoMann this should help

**Comment by DerManoMann** [comment]
Stripped it down to this - you really need bot the `preg_replace` **and** the `array_map`! Now for the fun part...
```
    public function whatever(string $key = 'xx')
    {
        preg_replace("/:${key}/", 'y', 'abx');

        array_map(static function ($issue) use ($key) {
            return $issue;
        }, []);
    }
```

**Comment by DerManoMann** [comment]
There is always one more token to consider... Updated PR #1017 ready for some testing @fuale .

**Comment by DerManoMann** [comment]
Fixed in 4.0.2
